apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx

```diff
@@ -0,0 +1,494 @@
+  }, [continueToSelectDelegatesStep, continueToAuthorizeTxStep]);
+
+  const closeModal = useCallback(() => {
+    setIsSubmitAndSignTxLoading(false);
+    setIsModalOpen(false);
+    setAmountToBond(0);
+    setPaymentDestination(PaymentDestination.Staked);
+    setSelectedValidators([]);
+    setDelegateTxStep(DelegateTxSteps.BOND_TOKENS);
+  }, [setIsModalOpen]);
+
+  const submitAndSignTx = useCallback(async () => {
+    setIsSubmitAndSignTxLoading(true);
+
+    try {
+      if (isFirstTimeNominator) {
+        const bondTxHash = await bondTokens(
+          walletAddress,
+          amountToBond,
+          PaymentDestination.Stash
+        );
+
+        if (bondTxHash) {
+          const bondTx = await evmPublicClient.waitForTransactionReceipt({
+            hash: bondTxHash,
+          });
+
+          if (bondTx.status === 'success') {
+            notificationApi({
+              variant: 'success',
+              message: `Successfully bonded ${amountToBond} tTNT.`,
+            });
+
+            const updatePaymentDestinationTxHash =
+              await updatePaymentDestination(walletAddress, paymentDestination);
+
+            if (updatePaymentDestinationTxHash) {
+              const updatePaymentDestinationTx =
+                await evmPublicClient.waitForTransactionReceipt({
+                  hash: updatePaymentDestinationTxHash,
+                });
+
+              if (updatePaymentDestinationTx.status === 'success') {
+                notificationApi({
+                  variant: 'success',
+                  message: `Successfully updated payment destination to ${paymentDestination}.`,
+                });
+
+                const nominateTxHash = await nominateValidators(
+                  walletAddress,
+                  selectedValidators
+                );
+
+                if (nominateTxHash) {
+                  const nominateTx =
+                    await evmPublicClient.waitForTransactionReceipt({
+                      hash: nominateTxHash,
+                    });
+
+                  if (nominateTx.status === 'success') {
+                    notificationApi({
+                      variant: 'success',
+                      message: `Successfully nominated ${selectedValidators.length} validators.`,
+                    });
+
+                    closeModal();
+                  }
+                }
+              } else {
+                notificationApi({
+                  variant: 'error',
+                  message: 'Failed to update payment destination!',
+                });
+
+                closeModal();
+              }
+            }
+          } else {
+            notificationApi({
+              variant: 'error',
+              message: 'Failed to bond tokens!',
+            });
+
+            closeModal();
+          }
+        }
+      } else {
+        if (amountToBond > 0) {
+          const bondTxHash = await bondExtraTokens(walletAddress, amountToBond);
+
+          if (bondTxHash) {
```

Comment:
This is a tremendous amount of nesting and very unclean/unreadable. Can you rework this? We should try to reduce the length of this logic in place of helper functions and less nesting.